### ISSUE TITLE
GcpFirestoreEmulatorAutoConfiguration doesn't seem to work with testcontainers

### ISSUE THREAD

**Issue by kamil-pociot** [GoogleCloudPlatform/spring-cloud-gcp#2286]
Hi, following documentation I am trying to run integration test that would save User entity to Firestore running on testcontainer. Test is written on a freshly downloaded project from start.spring.io. Project runs on java 17, spring boot 3.1.5, gcp-cloud v4.8.2, and spring cloud 2022.0.4. 

Problem:
Whenever I try to use Firestore service from spring context or try to save entity with FirestoreReactiveRepository I get following error:

```
io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Database name "projects/test-project/databases/(default)/documents" has extra characters at index 41.
```

However test works if I prepare new instance of Firestore service with modified host and credentials.

Am I getting this wrong or should `GcpFirestoreEmulatorAutoConfiguration` configure all the required dependecies to work with testcontainer based emulator?

I prepared minimal project to reproduce the issue:
https://github.com/kamil-pociot/gcp-demo

I'd appreciate any help. Thanks!

**Comment by YoHanGLoot** [comment]
Can add that I also tried this and got the same error. I have had not the pleasure yet of finding any solutions myself

**Comment by asattelmaier** [comment]
I'm experiencing the same issue. It seems to work as expected up to `spring-cloud-gcp-starter-firestore` version `3.6.3`, but starting from version `3.7.0` I'm encountering this error as well.

**Comment by henrikm-cdw** [comment]
I'm having the exact same issue with `spring-cloud-gcp-starter-firestore` version `4.8.4`
This basically makes unit testing of Firestore with TestContainers impossible. Is there a plan to fix this anytime soon? Is there a workaround? Thanks.

**Comment by zhumin8** [comment]
Fix for #2290 (#2348, #2429) merged in and included in releases [v3.7.6](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases/tag/v3.7.6) and [v4.9.0](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases/tag/v4.9.0). All reports above are using older versions. 

Please try with newer version and reopen if this is still a problem.
